### ISSUE TITLE
-di and -dg aren't grabbing anything with this latest release (win x86-64)

### ISSUE THREAD

**Issue by Mincka** [Mincka/DMArchiver#17]

**Comment by Gorrrg** [comment]
I think #16 is related. I think the problem is with the #14 fix. Text messages are put under `DirectMessage-contentContainer` but all other types of messages are still booked under `DirectMessage-container`? After the fix only the class `DirectMessage-contentContainer` is crawled for elements.

**Comment by Gorrrg** [comment]
Ok, so the hierarchy is as follows. `DirectMessage-container` is the parent, and under it there's elements like `DirectMessage-media u-borderRadiusInherit u-chromeOverflowFix` for images and `DirectMessage-contentContainer` specifically for the text element `DirectMessage-text`.

I think the best solution would be to roll the code back to before the #14 fix, and in the for-loop instead of matching `DirectMessage-text` match for `DirectMessage-contentContainer`. If that is matched create another sub for-loop that verifies there's a `DirectMessage-text` in it (or else it's "Unknown element").

**Comment by Gorrrg** [comment]
No actually, come to think of it that won't work. Now DirectMessage-container can have multiple elements in one message that are of importance. For instance if an image was sent with a description.

**Comment by Mincka** [comment]
Thank you for looking into it. Indeed, I didn't check completely the proposed fix and few issues are still there.

As I can see now, we have:
- For a single text message:
```
DirectMessage
- DirectMessage-container
-- DirectMessage-message with-text
--- DirectMessage-contentContainer
---- DirectMessage-text
```

- For a single image without caption:
```
DirectMessage
- DirectMessage-container
-- DirectMessage-message with-media
--- DirectMessage-attachmentContainer
---- DirectMessage-media
----- Sensitive
------- Sensitive-content
-------- Media
--------- Media-photo
--- DirectMessage-contentContainer (empty in this case)
```

- For a tweet:
```
DirectMessage
- DirectMessage-container
-- DirectMessage-message with-tweet
--- DirectMessage-attachmentContainer
---- DirectMessage-tweet
----- QuoteTweet
------- QuoteTweet-container
-------- + a few more
--- DirectMessage-contentContainer (empty in this case)
```

- For a video (such as https://t.co/B3PY7Dk1Fq)
```
DirectMessage
- DirectMessage-container
-- DirectMessage-message with-card
--- DirectMessage-attachmentContainer
---- DirectMessage-card
--- DirectMessage-contentContainer (empty in this case)
```

- For a GIF (embedded video)
```
DirectMessage
- DirectMessage-container
-- DirectMessage-message with-media
--- DirectMessage-attachmentContainer
---- DirectMessage-media
----- Sensitive
------ Sensitive-content
------- Media
-------- Media-video
--------- PlayableMedia PlayableMedia--gif
---------- PlayableMedia-player (with preview URL)
--- DirectMessage-contentContainer (empty in this case)
```

- For a an embedded video (uploaded video)
```
DirectMessage
- DirectMessage-container
-- DirectMessage-message with-media
--- DirectMessage-attachmentContainer
---- DirectMessage-media
----- Sensitive
------ Sensitive-content
------- Media
-------- Media-video
--------- PlayableMedia PlayableMedia--video
---------- PlayableMedia-player (with preview URL)
--- DirectMessage-contentContainer (empty in this case)
```

- For a single sticker
```
DirectMessage
- DirectMessage-container
-- DirectMessage-message
--- DirectMessage-media
---- Media
----- Media-photo
------ FlexEmbed-item (for the link and description)
--- DirectMessage-contentContainer (empty in this case)
```

**Comment by Mincka** [comment]
It seems I've managed to make a proper fix without too much effort.

Could you test this release before I push the commit please? Again thanks a lot for your help and your interest in the project. It is a motivation to improve and fix the tool. 😃 

[dmarchiver-0.1.3-windows-x86-64.zip](https://github.com/Mincka/DMArchiver/files/1099537/dmarchiver-0.1.3-windows-x86-64.zip)

**Comment by goofyhsk** [comment]
di and dg looking great now

I've noticed originally (I started using with 0.1.1) I had 38 conversation txts, now I only get 19 with 0.1.2, 0.1.3

(I count about 37 on the twitter site still)

**Comment by Mincka** [comment]
Thanks for the feedback. Another interesting issue. Are the retrieved conversations the 19 most recent ones? This may be related to #8 but starting with 20 conversations instead of 50.

**Comment by goofyhsk** [comment]
Can confirm that, 19 most recent.

**Comment by sussron** [comment]
Hi Julien,I pasted this on the Fix #17 but realize it's closed and perhaps
you wont' see it. so maybe i should email the issue here. I wasn't sure, so
i'm sorry for sending it to you twice.:

Hi Julien,
I've downloaded the most recent release 1.3 as I hadn't backed up messages
since February. I decided to do just one of the conversations with images.
I i put in the commands as you showed me (remember I'm not the techie one
here so i apologize for my ignorance) . i'm still getting the Unknown
element error that other users mentioned in prior threads. it is still
processing tweets, but what is it not including so that I'm sure what i'm
getting? Thanks in advance for your help

Thanks,
Ronnie

PS. the resulting txt file had pulled 175000 messages but it's just the
date and username and not the actual text messages that were typed.

**Comment by sussron** [comment]
And julien,
I compared it to the file from February which was 18.5MB whereas this empty
one is 6.5MB in size. the images folder has only 15 stickers in it and not
the actual message images. I'm attaching screen shots for you to compare.
Also the date of the message goes back to 8/5/15 as the attachment shows,
but the file that came this time started 12/5/15.

Thanks so much!
Ronnie
